core/jvm/src/main/scala/scalaz/zio/RTS.scala

```diff
@@ -924,25 +900,32 @@ private object RTS {
       io.ensuring(exitUninterruptible)
     }
 
-    final def register(cb: Callback[E, A]): Async[E, A] = join0(cb)
+    final def register(cb: Callback[E, A]): Async[E, A] =
+      observe0 {
+        case ExitResult.Completed(r) => cb(r)
+        case ExitResult.Failed(e, _) => e
+        case ExitResult.Terminated(ts) =>
+          rts.submit(rts.unsafeRun(unhandled(ts)))
+          cb(ExitResult.Terminated(Errors.TerminatedFiber :: Nil))
```

Comment:
Why not use `ts` here, when invoking `cb`?

Reply:
I thought that branch represents a catastrophic bug in RTS, not in user code. Perhaps there could be a special error that when printed asks the user to submit a bug report. I may be thinking too hard here 😅

Reply:
IMHO it's better to preserve them, because if someone runs into a catastrophic bug they'll need the debug info